lib/galaxy/managers/notification.py

```diff
@@ -0,0 +1,75 @@
+from typing import (
+    List,
+    Optional,
+)
+
+from sqlalchemy import false
+
+from pydantic import (
+    BaseModel,
+    Field,
+)
+
+from galaxy import model
+from galaxy.exceptions import ObjectNotFound
+from galaxy.model.scoped_session import galaxy_scoped_session
+from galaxy.structured_app import MinimalManagerApp
+
+class NotificationManager:
+    """Interface/service object shared by controllers for interacting with notifications."""
+
+    def __init__(self, app: MinimalManagerApp, sa_session: galaxy_scoped_session):
+        self._app = app
+        self.sa_session = sa_session
+
+    def index(self, limit: int = 10, offset: int = 0, user: Optional[model.User] = None):
+        """
+        Displays a collection (list) of notifications.
+        """
+        rval = []
+        for notification in (
+            self.sa_session.query(model.Notification)
+            .filter(model.Notification.deleted == false())
+            .limit(limit)
+            .offset(limit + offset)
+        ):
+            rval.append(notification)
+        return rval
+
+    def create(self, trans, payload):
```

Comment:
You do not need trans here, you can use self.sa_session. You should also add the type annotation for payload here.